src/rospack.cpp

```diff
@@ -1590,14 +1592,15 @@ Rosstackage::computeDeps(Stackage* stackage, bool ignore_errors, bool ignore_mis
 void
 Rosstackage::computeDepsInternal(Stackage* stackage, bool ignore_errors, const std::string& depend_tag, bool ignore_missing)
 {
-  TiXmlElement* root;
+  XMLElement* root;
   root = get_manifest_root(stackage);
 
-  TiXmlNode *dep_node = NULL;
   const char* dep_pkgname;
-  while((dep_node = root->IterateChildren(depend_tag, dep_node)))
+  for(XMLElement *dep_node = root->FirstChildElement(depend_tag.c_str());
+      dep_node;
```

Comment:
It's not statement, but an expression checking the loop's exhaustion before dereferencing the pointer in the next line.

But I see now that the type casting in the line 1603 is redundant. Will update.